includes/Pages/PageSearch.php

```diff
@@ -27,112 +29,167 @@ protected function main()
-                $this->redirect("search");
+
+                $this->assign('term', $searchTerm);
+                $this->assign('target', $searchType);
+                $this->assign('excludeNonConfirmed', $excludeNonConfirmed);
+                $this->assign('hasResultset', false);
 
                 return;
             }
 
-            $results = array();
+            // searchType known to be sane from the validate step above
+            if (!$this->barrierTest('by' . ucfirst($searchType), User::getCurrent($this->getDatabase()))) {
+                // only accessible by url munging, don't care about the UX
+                throw new AccessDeniedException($this->getSecurityManager(), $this->getDomainAccessManager());
+            }
+
+            $requestSearch = RequestSearchHelper::get($database);
+
+            $this->setSearchHelper($requestSearch);
+            $this->setupLimits();
+
+            if ($excludeNonConfirmed) {
+                $requestSearch->withConfirmedEmail();
+            }
 
             switch ($searchType) {
                 case 'name':
-                    $results = $this->getNameSearchResults($searchTerm);
+                    $this->getNameSearchResults($requestSearch, $searchTerm);
                     break;
                 case 'email':
-                    $results = $this->getEmailSearchResults($searchTerm);
+                    $this->getEmailSearchResults($requestSearch, $searchTerm);
                     break;
                 case 'ip':
-                    $results = $this->getIpSearchResults($searchTerm);
+                    $this->getIpSearchResults($requestSearch, $searchTerm);
                     break;
+                case 'comment':
+                    $this->getCommentSearchResults($requestSearch, $searchTerm);
+                    break;
+            }
+
+            /** @var Request[] $results */
+            $results = $requestSearch->getRecordCount($count)->fetch();
+
+            $formParameters = [
+                'term' => $searchTerm,
+                'type' => $searchType,
+            ];
+
+            if ($excludeNonConfirmed) {
+                $formParameters['excludeNonConfirmed'] = true;
             }
 
+            $this->setupPageData($count, $formParameters);
+
             // deal with results
             $this->assign('requests', $this->prepareRequestData($results));
             $this->assign('resultCount', count($results));
-            $this->assign('term', $searchTerm);
-            $this->assign('target', $searchType);
+            $this->assign('hasResultset', true);
 
             list($defaultSort, $defaultSortDirection) = WebRequest::requestListDefaultSort();
             $this->assign('defaultSort', $defaultSort);
             $this->assign('defaultSortDirection', $defaultSortDirection);
-
-            $this->assignCSRFToken();
-            $this->setTemplate('search/searchResult.tpl');
         }
         else {
-            $this->assignCSRFToken();
```

Comment:
As far as I can tell, the new search form doesn't have any CSRF protection. Was that intentional?

Reply:
🤦‍♂️ Okay I'm just dumb, I saw that it was using GET and just didn't actually connect that knowledge to anything.